app/Services/ScrapeService.php

```diff
@@ -0,0 +1,227 @@
+
+                $submit = $this->client->waitFor('[data-testid="search-button-with-loader"]');
+                // First click to confirm choice, Second to redirect
+                $submit->filter('[data-testid="search-button-with-loader"]')
+                    ->click();
+                $submit->filter('[data-testid="search-button-with-loader"]')
+                    ->click();
+            } else {
+                $url = $this->client->getCurrentURL() . ';pa-' . $page;
+                $this->client->request('GET', $url);
+            }
+
+            $this->client->waitFor('[data-testid="accommodation-list"]', config('scraper.crawler.timeouts.long'));
+
+            for ($i = 1; $i < config('scraper.crawler.hotels_per_page'); $i++) {
+                if ($count == $quantity)
+                    die();
+
+                try {
+                    $building = $this->fetchBuilding($i);
+                    $this->appendToFiles($building, $directoryName);
+
+                    $count++;
+                } catch (Exception) {
+                    echo "Ommitted a building: Failed to scrape building $i on page $page.\n";
+                }
+            }
+            $page++;
+        }
+    }
+
+    /**
+     * @throws NoSuchElementException
+     * @throws TimeoutException
+     * @throws Exception
+     */
+    private function fetchBuilding(int $id): Building
+    {
+        $buildingSelector = "[data-testid='accommodation-list-element']:nth-of-type($id)";
+        $timeout = config('scraper.crawler.timeouts.short');
+
+        $this->client->executeScript("document.querySelector(\"$buildingSelector\").scrollIntoView(true)");
+
+        $buildingElement = $this->client->waitFor($buildingSelector, $timeout)->filter($buildingSelector);
+
+        // Show photos panel
+        $buildingElement->filter("button")->click();
+
+        // Hotel name
+        $name = $this->fetchBuildingName($buildingElement);
+        // Hotel / hostel / resort etc.
+        $type = $this->fetchBuildingType($buildingElement);
+        $city = $this->fetchBuildingCity($buildingElement);
+
+        // Main image
+        $mainImage = $buildingElement->filter("[data-testid=\"accommodation-main-image\"]")->attr('src');
+
+        $this->client->waitFor("$buildingSelector [data-testid=\"grid-gallery\"]", $timeout);
+        $images = $this->fetchBuildingImages($buildingElement);
+
+        // Opens info panel
+        $buildingElement->filterXPath('.//button[contains(text(), "Info")]')->click();
+
+        // Description
+        $this->client->waitFor("$buildingSelector [data-testid=\"info-slideout\"]", $timeout);
+        $body = $this->fetchBuildingBody($buildingElement);
+        $street = $this->fetchBuildingAddress($buildingElement);
+
+        // More amenities button
+        $buildingElement->filter("[data-testid=\"toggle-all-amenities\"]")->click();
+
+        $amenities = $this->fetchBuildingAmenities($buildingElement);
```

Comment:
Those comments still make no sense. I see no difference between this and the last review.